File: Api/src/main/java/net/binggl/mydms/features/documents/models/ActionResult.java
package net.binggl.mydms.features.documents.models;

public enum ActionResult {
	None(0, "None"),
	Created(1, "Saved"),
	Updated(2, "Found"),
	Deleted(3, "Deleted"),
	Error(99, "Error");
		
	private Integer id;
    private String name;
 
    private ActionResult(final Integer id, final String name) {
        this.id = id;
        this.name = name;
    }

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
}

File: Api/src/main/kotlin/net/binggl/mydms/shared/models/ActionResult.kt
package net.binggl.mydms.shared.models

enum class ActionResult constructor(private val id: Int, private val resultName: String) {
    None(0, "None"),
    Created(1, "Created"),
    Updated(2, "Updated"),
    Deleted(3, "Deleted"),
    Error(99, "Error")
}